programs/ssl/ssl_client2.c

```diff
@@ -905,6 +927,62 @@ int main( int argc, char *argv[] )
     }
 #endif /* MBEDTLS_KEY_EXCHANGE__SOME__PSK_ENABLED */
 
+#if defined(MBEDTLS_ECP_C)
+    if( opt.curves != NULL )
+    {
+        p = (char *) opt.curves;
+        i = 0;
+
+        if( strcmp( p, "none" ) == 0 )
+        {
+            curve_list[0] = MBEDTLS_ECP_DP_NONE;
+        }
+        else if( strcmp( p, "default" ) != 0 )
+        {
+            /* Leave room for a final NULL in curve list */
+            while( i < (int) sizeof(curve_list) - 1 && *p != '\0' )
```

Comment:
I think you can only guarantee that the while loop condition will work if `sizeof(mbedtls_ecp_group_id)` is 1 byte. Might be a bit fragile.